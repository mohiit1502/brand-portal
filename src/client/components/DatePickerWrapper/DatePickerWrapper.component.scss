.c-DatePickerWrapper {

  &:not(:last-child) {
    margin-bottom: 35px;
  }

  .SingleDatePicker, .SingleDatePickerInput, .DateInput {
    width: 100%;
  }

  .SingleDatePickerInput {
    height: 34px;
    border: none;
    color: #0b1e40;

    .DateInput_input {
      font-size: 0.875rem;
      line-height: 1.25rem;
      //color: #2c333a;
      cursor: pointer;
      padding: 0.375rem 0.75rem 0.375rem 1.25rem;
      border: solid 1px #006799;
      border-radius: 42px;

      &.DateInput_input__focused {
        box-shadow: 0 3px 18px 0 rgba(122, 196, 228, 0.2);
        border: solid 1px #7ac4e4;
      }

    }

    .SingleDatePickerInput_calendarIcon {
      background-image: url("./../../images/calender.png");
      background-repeat: no-repeat;
      position: absolute;
      top: 0.4rem;
      right: 0.5rem;
      width: 1rem;
    }
  }

  .SingleDatePicker_picker {
    top: 37px !important;
    left: 1px !important;
    z-index: 8;
    width: 100%;
  }

  .CalendarMonthGrid {
    left: 0;
    .CalendarMonth {
      padding-left: 2.5rem !important;
      padding-right: 2.5rem !important;

      .CalendarMonth_caption {
        font-size: 0.875rem !important;
      }

      td.CalendarDay {
        font-size: 0.875rem;
        &.CalendarDay__selected {
          background: #017dc6;
          border-radius: 50%;
          display: block;
          height: 35px !important;
          margin-left: auto;
          margin-right: auto;
          padding-top: 8px;
          max-width: 36px !important;
        }
      }
    }
  }

  .DayPicker, .DayPicker > div, .DayPicker > div > div, .DayPicker_transitionContainer, .CalendarMonthGrid, .CalendarMonthGrid_month__horizontal, .CalendarMonth_table, .DayPicker_weekHeader {
    width: 100% !important;
  }

  .DayPicker_weekHeader {
    padding-left: 2.5rem !important;
    padding-right: 2.5rem !important;
  }

  .DayPicker_weekHeaders__horizontal {
    margin-left: 0;

    .DayPicker_weekHeader_li {
      font-size: 0.875rem;
      line-height: 1.25rem;
      color: black;
      width: 3.3rem !important;
    }
  }

  .DayPickerNavigation_leftButton__horizontalDefault {
    padding-left: 2.2rem;
  }

  .DayPickerNavigation_rightButton__horizontalDefault {
    padding-right: 2.2rem;
  }

  .custom-input-label {
    color: #0b1e40;
    position: absolute;
    top: 0.5rem;
    left: 1.5rem;
    background: white;
    font-size: 0.875rem;
    line-height: 1.25rem;
    margin-bottom: 0;
  }

  .shiftup-custom-input-label {
    position: absolute;
    z-index: 7;
    margin: 0;
    top: -10px;
    left: 1.5rem;
    background: white;
    font-size: 12px;
    -webkit-transition: top 100ms, font-size 100ms, background 100ms;
    transition: top 100ms, font-size 100ms, background 100ms;
    color: #0b1e40;

    span {
      margin-left: 5px;
      margin-right: 5px;
    }
  }

  .DayPickerNavigation_button__default {
    border: none;
  }

  .text-black {
    color: #0b1e40;
  }
}
