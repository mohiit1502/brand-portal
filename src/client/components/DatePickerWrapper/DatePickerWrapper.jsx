import React, {useEffect, useState} from 'react';
import PropTypes from 'prop-types';

import "react-dates/initialize";
import {SingleDatePicker} from "react-dates";
import {isInclusivelyAfterDay, isInclusivelyBeforeDay} from "react-dates/esm";
import moment from "moment";
import {SingleDatePickerPhrases} from "react-dates/esm/defaultPhrases";
import {Event} from "./../../images";
import './DatePickerWrapper.component.scss';

const defaultProps = {
  // example props for the demo
  autoFocus: false,
  initialDate: null,

  // input related props
  id: 'date',
  // placeholder: 'Select Date',
  disabled: false,
  required: false,
  screenReaderInputMessage: '',
  showClearDate: false,
  showDefaultInputIcon: false,
  customInputIcon: <p></p>,
  inputIconPosition: "after",
  block: false,
  small: false,
  regular: false,
  verticalSpacing: undefined,
  keepFocusOnInput: false,

  // calendar presentation and interaction related props
  renderMonthText: null,
  orientation: "horizontal",
  anchorDirection: "left",
  horizontalMargin: 0,
  withPortal: false,
  withFullScreenPortal: false,
  initialVisibleMonth: null,
  numberOfMonths: 1,
  keepOpenOnDateSelect: false,
  reopenPickerOnClearDate: false,
  isRTL: false,

  // navigation related props
  navPrev: null,
  navNext: null,
  onPrevMonthClick() {},
  onNextMonthClick() {},
  onClose() {},

  // day presentation and interaction related props
  renderCalendarDay: undefined,
  renderDayContents: null,
  enableOutsideDays: false,
  isDayBlocked: () => false,
  // isOutsideRange: day => !isInclusivelyAfterDay(day, moment()),
  isOutsideRange: day => !isInclusivelyBeforeDay(day, moment()),
  isDayHighlighted: () => {},

  hideKeyboardShortcutsPanel: true,

  // internationalization props
  displayFormat: () => moment.localeData().longDateFormat('LL'),
  monthFormat: 'MMMM YYYY',
  phrases: SingleDatePickerPhrases,
  weekDayFormat: 'ddd'
};

const DatePickerWrapper = props => {
  const {date, setDate} = props;
  const [focused, setFocused] = useState(false);
  const label = focused || date ? props.focusedLabel : props.label;
  let isOutsideRange = props.isOutsideRange;
  if (props.startDate) {
    isOutsideRange = day => !isInclusivelyBeforeDay(day, moment()) || !isInclusivelyAfterDay(day, props.startDate);
  } else if (props.endDate) {
    isOutsideRange = day => !isInclusivelyBeforeDay(day, props.endDate);
  }

  return (
    <div className="c-DatePickerWrapper row mt-12">
      <div className="col">
        <SingleDatePicker
          {...props}
          placeholder=""
          date={date}
          focused={focused}
          isOutsideRange={isOutsideRange}
          onDateChange={(date) => setDate(date)}
          onFocusChange={({focused}) => setFocused(focused)}
        />
        <label className={`custom-input-label custom-input-label-placeholder${focused || date ? " shiftup-custom-input-label" : ""}${!focused && date ? " text-black" : ""}`} htmlFor={props.id}>
          <span className="label-text"> {label} </span>
        </label>
      </div>
    </div>
  );
};

DatePickerWrapper.propTypes = {
  date: PropTypes.object,
  setDate: PropTypes.func
};

DatePickerWrapper.defaultProps = defaultProps;

export default DatePickerWrapper;
