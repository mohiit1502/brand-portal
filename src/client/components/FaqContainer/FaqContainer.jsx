import React from "react";
import PropTypes from "prop-types";
import "./FaqContainer.component.scss";

const FaqContainer = props => {
  return (
    <div className="c-FaqContainer">
      In Component FaqContainer
    </div>
  );
};

FaqContainer.propTypes = {

};

export default FaqContainer;