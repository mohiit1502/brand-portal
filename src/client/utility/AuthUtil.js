import authConfig from "./../config/authorizations";
import Helper from "./helper";

export default class AuthUtil {
  static isActionAccessible (action, userProfile, actedOnRole) {
    if (action && action.indexOf(".") > -1) {
      const actionConfig = Helper.search(action, authConfig);
      if (actionConfig && userProfile && userProfile.role && userProfile.role.name && typeof actionConfig === "object") {
        const role = userProfile.role.name;
        if (actionConfig.length !== undefined) {
          return actionConfig.includes(role);
        } else {
          if (Object.keys(actionConfig).includes(role) && actedOnRole && actionConfig[role].includes(actedOnRole)) {
            return true;
          }
        }
      }
    }
    return false;
  }
}
