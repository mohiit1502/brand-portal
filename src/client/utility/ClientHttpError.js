export default class ClientHttpError {

  status;
  error;
  message;

  constructor (status, error, message) {
    this.status = status;
    this.error = error;
    this.message = message;
  }

}
