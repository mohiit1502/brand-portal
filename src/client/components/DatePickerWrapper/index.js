import DatePickerWrapper from './DatePickerWrapper.jsx';

export default DatePickerWrapper;