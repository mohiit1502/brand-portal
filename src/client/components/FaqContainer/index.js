import FaqContainer from "./FaqContainer.jsx";

export default FaqContainer;
