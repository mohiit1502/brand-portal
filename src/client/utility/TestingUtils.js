import {applyMiddleware, compose, createStore} from "redux";
import rootReducer from "../reducers";
import thunk from "redux-thunk";
import Immutable from "immutable";
import configureMockStore from 'redux-mock-store';
import ClientHttpError from "./ClientHttpError";

export const findByTestAttribute = (component,attr) => component.find(`[data-test="${attr}"]`);

export const testStore = state => configureMockStore()(state);

export const clearKeys = (tree, arr) => {
  if (arr && arr.indexOf(tree) > -1) {
    return;
  }
  if (tree && typeof tree === 'object') {
    if (tree.inputId) {
      return;
    }
    arr.push(tree)
    if (tree.key) {
      try {
        delete tree.key;
      } catch (e) {}
    }
    if (Array.isArray(tree)) {
      tree.forEach(obj => clearKeys(obj, arr));
    } else {
      Object.keys(tree).forEach(key => clearKeys(tree[key], arr));
    }
  }
}

export function setupFetchStub(data) {
  return function fetchStub(_url) {
    return new Promise((resolve) => {
      resolve({
        json: () =>
          Promise.resolve({
            data,
          }),
      })
    })
  }
}

export function setupFetchThrowStub() {
  const error = new ClientHttpError(500, "test-error"); // Directly throwing new ClientHttpError fails in Jest
  throw error;
}
