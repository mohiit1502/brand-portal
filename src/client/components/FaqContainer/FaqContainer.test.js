/* eslint-disable filenames/match-regex, no-unused-vars, no-undef */
import React from "react";
import FaqContainer from "./FaqContainer";
import {mount} from "enzyme";
import DatePickerWrapper from "../DatePickerWrapper";
import moment from "moment";
import toJson from "enzyme-to-json";
import PropTypes from "prop-types";
import {testStore} from "../../utility/TestingUtils";

const setUp = (props) => {

  return mount(props ? <FaqContainer /> : <FaqContainer />);
};

jest.mock("react", () => {
  const originReact = jest.requireActual("react");
  const mUseRef = jest.fn();
  return {
    ...originReact,
    useRef: mUseRef
  };
});

describe("FaqContainer", () => {
  it("renders without error", () => {
    let wrapper;
    beforeEach(() => {
      jest.restoreAllMocks();
      jest.resetAllMocks();
    });

    wrapper = setUp();

    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();

  });
});
