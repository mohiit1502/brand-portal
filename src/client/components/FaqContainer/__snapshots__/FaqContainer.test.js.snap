// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqContainer renders without error 1`] = `
<FaqContainer>
  <div
    className="c-FaqContainer"
  >
    In Component FaqContainer
  </div>
</FaqContainer>
`;
